api/tests/Behat/bootstrap/v2/Common/ExpectedResultsTrait.php

```diff
@@ -6,83 +6,113 @@
-                    }
-                } else {
-                    $xpath = '//td|//th';
-                    $tableDataItems = $tableItem->findAll('xpath', $xpath);
-                    $this->getTextItems($tableDataItems);
-                }
+        foreach ($summarySectionElements as $summarySectionElement) {
+            $this->summarySectionItemsFound = [];
+            if ('dl' == $summarySectionElement->getTagName()) {
+                $this->updateSummarySectionItemsFoundFromDescriptionList($summarySectionElement);
+            } elseif ('tbody' == $summarySectionElement->getTagName()) {
+                $this->updateSummarySectionItemsFoundFromTableBody($summarySectionElement);
             } else {
-                $this->throwContextualException('Unrecognised option');
+                $this->throwContextualException('Element must be either dl or tbody');
             }
-            $sections[] = $this->comparisonItems;
+            $sections[] = $this->summarySectionItemsFound;
         }
 
-//        ** Uncomment below for debugging purposes... useful when creating your tests**
-//        foreach ($sections as $sectionKey=>$section) {
-//            var_dump("Section key: ".strval($sectionKey));
-//            var_dump($section);
-//        }
-//        var_dump($expectedResultArrays);
+        if ($debug) {
+            $this->debugExpectedResultsDisplayed($sections, $summarySectionNumber, $expectedResults);
+        }
 
-        foreach ($sections[$sectionNumber] as $foundResultKey => $foundResults) {
-            $this->checkContainsExpectedResults($expectedResultArrays[$foundResultKey], $foundResults, $sectionNumber, $context);
+        foreach ($sections[$summarySectionNumber] as $foundResultKey => $foundResults) {
+            $this->checkContainsExpectedResults($expectedResults[$foundResultKey], $foundResults, $summarySectionNumber, $context);
         }
     }
 
-    private function getTextItems($items)
+    private function updateSummarySectionItemsFoundFromListRow($descriptionList)
+    {
+        $xpath = '//li';
+        $listItems = $descriptionList->findAll('xpath', $xpath);
+
+        if (count($listItems) > 0) {
+            $this->updateSummarySectionItemsFound($listItems);
+        } else {
+            $xpath = '//dt|//dd';
+            $descriptionDataItems = $descriptionList->findAll('xpath', $xpath);
+            $this->updateSummarySectionItemsFound($descriptionDataItems);
+        }
+    }
+
+    private function updateSummarySectionItemsFoundFromNoListRow($descriptionList)
+    {
+        $xpath = '//li';
+        $listItems = $descriptionList->findAll('xpath', $xpath);
+
+        if (count($listItems) > 0) {
+            $this->updateSummarySectionItemsFound($listItems);
+        } else {
+            $xpath = '//dt|//dd';
+            $descriptionDataItems = $descriptionList->findAll('xpath', $xpath);
+            $this->updateSummarySectionItemsFound($descriptionDataItems);
+        }
+    }
```

Comment:
Aren't these doing the same thing?

Reply:
they weren't when i first set it up and i seem to have made them do the same. it will actually allow more fine grained control if i split them out properly so done that :)